src/v/storage/api.cc

```diff
@@ -114,6 +114,10 @@ ss::future<> api::monitor() {
               "Log storage usage {} exceeds configured max {}",
               human::bytes(usage.usage.total()),
               human::bytes(_log_storage_max_size().value()));
+
+            // wake up housekeeping on every core to run garbage collection
+            co_await container().invoke_on_all(
+              [](auto& api) { api._log_mgr->trigger_housekeeping(); });
```

Comment:
Current "knobs" for dealing with reaching the maximum size are

1. Blocking new kafka produces
2. Aggressively running GC

The third knob will be removing data in violation of retention settings when that data has been uploaded into the cloud already. https://github.com/redpanda-data/core-internal/issues/268